doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-Linux/index.md

```diff
@@ -2,304 +2,198 @@
-    1. For automatic installation, execute the following commands (tested on Ubuntu 14.04):
+            error = kaa_client_start(kaa_client, &dummy_function, (void *)kaa_client, 0);
 
-       ~~~
-       $ sudo apt-get install libsqlite3-0 libsqlite3-dev
-       ~~~
+            kaa_client_destroy(kaa_client);
 
-    2. For manual installation, refer to the following example:
+            if (error) {
+                return -1;
+            }
 
-       ~~~
-       $ wget https://www.sqlite.org/2015/sqlite-autoconf-3081002.tar.gz
-       $ tar -zxf sqlite-autoconf-3081002.tar.gz
-       $ cd sqlite-autoconf-3081002/
-       $ ./configure
-       $ sudo make install
-       ~~~
+            return 0;
+        }
 
-**NOTE:** Instead of manually installing all required components and libraries, you can follow [the quick way to build C/C++ endpoint SDK](#Linux-Quickway).  (only applicable for x86\_64 platform build)
+    The application code is simple: application just prints *KAA_APP_PRINT_TIMES* times the string *"Hello, I am a Kaa Application!\n"*. That's it. So, now we have the following file structure:
 
-## Compile C++ endpoint SDK
+        - my-kaa-application
+           - kaa
+             - Unpacked C SDK
+           - src
+             - kaa-application.c
 
-To build the C++ endpoint SDK, do the following:
+5. Create CMakeLists.txt file in the my-kaa-application directory. It is a top-level cmake file which is responsible for the application build.
 
-<!-- TODO: Fix this link when Administration-UI-guide#AdministrationUIguide-GeneratingSDK are updated -->
-1. [Generate](Administration-UI-guide#AdministrationUIguide-GeneratingSDK) the C++ endpoint SDK in Admin UI.
-2. Download and untar the Kaa C++ SDK archive.
+        touch CMakeLists.txt
 
-   ~~~ bash
-   $ tar xfv kaa-cpp-ep-sdk.tar.gz
-   ~~~
+6. Add the following code to top-level CMakeLists.txt:
 
-3. Run the following commands.
+        cmake_minimum_required(VERSION 2.8.8)
+        project(kaa-application C)
 
-   ~~~
-   mkdir build
-   cd build
-   cmake ..
-   make
-   make install
-   ~~~
+        find_package(OpenSSL REQUIRED)
 
-## Configure C++ endpoint SDK
+        set(APP_NAME "kaa-app")
+        set(KAA_SDK_DIR "kaa")
 
-To configure the C++ endpoint SDK build, you can optionally specify the following parameters for the **cmake** command.
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g -Wall -Wextra")
 
-{% include csv-table.html csv_path='cmake-cpp-options.csv' table_id='table-cpp-cmake-options' width='100%' %}
+        set(APPLICATION_SOURCE_FILES src/kaa-application.c)
 
-The following example illustrates the build procedure for the debug build, with the INFO log level and disabled EVENTS feature and specified path to the folder Kaa will be installed in:
+        add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/${KAA_SDK_DIR} ${CMAKE_CURRENT_BINARY_DIR}/c_sdk_build_directory)
```

Comment:
Second argument isn't needed. `${CMAKE_CURRENT_SOURCE_DIR}` can be dropped as well.
`${KAA_SDK_DIR}` referenced once, so use `kaa` instead of it.